src/async_impl/client.rs

```diff
@@ -408,6 +411,12 @@ impl ClientBuilder {
         self
     }
 
+    /// Apply [`Response::error_for_status`] on every `Response` returned by the client.
```

Comment:
```suggestion
    /// Turns every response returned by the client into an error if the server returned an error. 
    ///
    /// Calls [`Response::error_for_status_ref`] on every `Response` returned by the client.
```

This seems insufficiently descriptive of how this feature works to appear in documentation. Perhaps add:
1. default setting
2. more information about how this works
3. an example